azure/services/managedclusters/spec_test.go

```diff
@@ -163,6 +165,67 @@ func TestParameters(t *testing.T) {
 				g.Expect(result).To(BeNil())
 			},
 		},
+		{
+			name:     "set Linux profile if SSH key is set",
+			existing: nil,
+			spec: &ManagedClusterSpec{
+				Name:            "test-managedcluster",
+				ResourceGroup:   "test-rg",
+				Location:        "test-location",
+				Tags:            nil,
+				Version:         "v1.22.0",
+				LoadBalancerSKU: "Standard",
+				SSHPublicKey:    base64.StdEncoding.EncodeToString([]byte("test-ssh-key")),
+				GetAllAgentPools: func() ([]azure.ResourceSpecGetter, error) {
+					return []azure.ResourceSpecGetter{
+						&agentpools.AgentPoolSpec{
+							Name:          "test-agentpool-0",
+							Mode:          string(infrav1.NodePoolModeSystem),
+							ResourceGroup: "test-rg",
+							Replicas:      int32(2),
+							AdditionalTags: map[string]string{
+								"test-tag": "test-value",
+							},
+						},
+					}, nil
+				},
+			},
+			expect: func(g *WithT, result interface{}) {
+				g.Expect(result).To(BeAssignableToTypeOf(containerservice.ManagedCluster{}))
+				g.Expect(result.(containerservice.ManagedCluster).LinuxProfile).To(Not(BeNil()))
+				g.Expect(result.(containerservice.ManagedCluster).LinuxProfile.SSH.PublicKeys).To(Not(BeNil()))
```

Comment:
Can we update this check to a more fine-grained one?
```suggestion
				g.Expect(result.(containerservice.ManagedCluster).LinuxProfile.SSH.PublicKeys).To(Equal(base64.StdEncoding.EncodeToString([]byte("test-ssh-key"))))
```